infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -708,4 +742,46 @@ public void shutdown() {
     private String getLocalHost() {
         return NodeLocator.parseString(serverContext.getLocalEndpoint()).getHost();
     }
+
+    private LockClient createLockClient(LockStateType initState) throws Exception {
+        LockConfig lockConfig = createLockConfig(serverContext.getPluginConfigFilePath());
+        LockClient lockClient =
+                new LockClient(logReplicationNodeId,
+                        lockConfig, runtime.get());
+
+        LockListener lockListener =
+                new LogReplicationLockListener(this);
+
+        LockDataTypes.LockId lockId = LockDataTypes.LockId.newBuilder()
+                .setLockGroup(lockConfig.getLockGroup())
+                .setLockName(lockConfig.getLockName())
+                .build();
+
+        lockClient.getLocks().computeIfAbsent(
+                lockId,
+                key -> new Lock(lockId, lockListener, lockClient.getClientContext(), lockConfig,
+                        initState));
+
+        return lockClient;
+    }
+
+
+    @VisibleForTesting
+    public void deregisterToLogReplicationLock() throws Exception {
+        LockClient lockClient = createLockClient(LockStateType.HAS_LEASE);
+        lockClient.deregisterInterest();
```

Comment:
What happens if you deregisterInterest which enforces the LEASE_EXPIRED event which somehow will cause a callback on our listener, but the real lock (the previous to this) acquires the lock at that moment (which will also cause a callback on our DiscoveryService).

Reply:
This is why we increase the timeouts via the configuration in tests to prevent some other tasks from acquiring the lock.